cmake/thirdparty/SetupMPI.cmake

```diff
@@ -40,30 +40,35 @@ if (ENABLE_FIND_MPI)
     # Merge found MPI info and remove duplication
     #-------------------
     # Compile flags
-    if (ENABLE_CUDA)
+    set(_c_flg ${MPI_C_${_mpi_compile_flags_suffix}})
+    if (_c_flag AND ENABLE_CUDA)
```

Comment:
This was previously creating generator expressions for empty compile flags. We are now guarding this.